### ISSUE TITLE
there is no dir of jobName=NISA_Chronos/jobTimestamp=_ 	at com.kazurayam.inspectus.core.internal.AbstractService.step5_backupLatest(AbstractService.java:112)

### ISSUE THREAD

**Issue by kazurayam** [kazurayam/VisualInspectionOfExcelAndPDF#15]
Using the materialstore-0.16.2-SNAPSHOT, with the "store" dir and the "store-backup" dir not there, I ran `Test Cases/NISA/main_Chrosos`, I got the following error

```
2023-05-01 20:37:13.024 WARN  c.k.m.base.reduce.DiffingMPGProcessor    - #stuffDiffByDiffer left Material was NULL_OBJECT. right={"jobName":"NISA_Chronos","jobTimestamp":"20230501_203617","id":"c3f192874d9840021cb003059ea1664faead3d7c","fileType":"csv","metadata":{"seq":{"key":"seq","value":"4","matchedByAster":true}, "sheet_index":{"key":"sheet_index","value":"7","matchedByAster":true}, "sheet_name":{"key":"sheet_name","value":"保険会社","matchedByAster":true}, "URL.host":{"key":"URL.host","value":"www.fsa.go.jp","matchedByAster":true}, "URL.path":{"key":"URL.path","value":"/policy/nisa/20180126-1/02.xlsx","matchedByAster":true}, "URL.port":{"key":"URL.port","value":"80","matchedByAster":true}, "URL.protocol":{"key":"URL.protocol","value":"https","matchedByAster":true}},"identification":"{\"seq\":\"4\", \"sheet_index\":\"7\", \"sheet_name\":\"保険会社\", \"URL.host\":\"www.fsa.go.jp\", \"URL.path\":\"/policy/nisa/20180126-1/02.xlsx\", \"URL.port\":\"80\", \"URL.protocol\":\"https\"}","relativeUrl":"NISA_Chronos/20230501_203617/objects/c3f192874d9840021cb003059ea1664faead3d7c.csv","diffability":"AS_TEXT"}
2023-05-01 20:37:13.244 INFO  c.k.m.base.reduce.DiffingMPGProcessor    - #process {"checked":false,"isBachelor":true,"diffRatio":0.0,"fileTypeExtension":"xlsx","queryOnMetadata":{"URL.host":"www.fsa.go.jp", "URL.path":"/policy/nisa/20180126-1/02.xlsx", "URL.port":"80", "URL.protocol":"https", "seq":"4"},"identification":"{\"URL.host\":\"www.fsa.go.jp\", \"URL.path\":\"/policy/nisa/20180126-1/02.xlsx\", \"URL.port\":\"80\", \"URL.protocol\":\"https\", \"seq\":\"4\"}","left":{"jobName":"_","jobTimestamp":"_","id":"0000000000000000000000000000000000000000","fileType":"","metadata":{},"identification":"{}","relativeUrl":"_/_/objects/0000000000000000000000000000000000000000.","diffability":"UNABLE"},"right":{"jobName":"NISA_Chronos","jobTimestamp":"20230501_203617","id":"ab9914889a9eba9e3c001b7a3b5c1dc03098c65a","fileType":"xlsx","metadata":{"seq":{"key":"seq","value":"4","matchedByAster":true}, "URL.host":{"key":"URL.host","value":"www.fsa.go.jp","matchedByAster":true}, "URL.path":{"key":"URL.path","value":"/policy/nisa/20180126-1/02.xlsx","matchedByAster":true}, "URL.port":{"key":"URL.port","value":"80","matchedByAster":true}, "URL.protocol":{"key":"URL.protocol","value":"https","matchedByAster":true}},"identification":"{\"seq\":\"4\", \"URL.host\":\"www.fsa.go.jp\", \"URL.path\":\"/policy/nisa/20180126-1/02.xlsx\", \"URL.port\":\"80\", \"URL.protocol\":\"https\"}","metadataURL":"https://www.fsa.go.jp/policy/nisa/20180126-1/02.xlsx","relativeUrl":"NISA_Chronos/20230501_203617/objects/ab9914889a9eba9e3c001b7a3b5c1dc03098c65a.xlsx","diffability":"UNABLE"},"diff":{"jobName":"_","jobTimestamp":"_","id":"0000000000000000000000000000000000000000","fileType":"","metadata":{},"identification":"{}","relativeUrl":"_/_/objects/0000000000000000000000000000000000000000.","diffability":"UNABLE"}}
2023-05-01 20:37:13.244 WARN  c.k.m.base.reduce.DiffingMPGProcessor    - #stuffDiffByDiffer left Material was NULL_OBJECT. right={"jobName":"NISA_Chronos","jobTimestamp":"20230501_203617","id":"ab9914889a9eba9e3c001b7a3b5c1dc03098c65a","fileType":"xlsx","metadata":{"seq":{"key":"seq","value":"4","matchedByAster":true}, "URL.host":{"key":"URL.host","value":"www.fsa.go.jp","matchedByAster":true}, "URL.path":{"key":"URL.path","value":"/policy/nisa/20180126-1/02.xlsx","matchedByAster":true}, "URL.port":{"key":"URL.port","value":"80","matchedByAster":true}, "URL.protocol":{"key":"URL.protocol","value":"https","matchedByAster":true}},"identification":"{\"seq\":\"4\", \"URL.host\":\"www.fsa.go.jp\", \"URL.path\":\"/policy/nisa/20180126-1/02.xlsx\", \"URL.port\":\"80\", \"URL.protocol\":\"https\"}","metadataURL":"https://www.fsa.go.jp/policy/nisa/20180126-1/02.xlsx","relativeUrl":"NISA_Chronos/20230501_203617/objects/ab9914889a9eba9e3c001b7a3b5c1dc03098c65a.xlsx","diffability":"UNABLE"}
2023-05-01 20:37:13.255 DEBUG c.k.m.base.reduce.DiffingMPGProcessor    - #process stuffedMaterialProductList.size()=44
2023-05-01 20:37:13.260 INFO  c.k.k.c.keyword.builtin.CommentKeyword   - step3_reduceChronos finished
2023-05-01 20:37:13.263 INFO  c.k.k.c.keyword.builtin.CommentKeyword   - step4_report started
2023-05-01 20:37:13.799 INFO  c.k.k.c.keyword.builtin.CommentKeyword   - step4_report finished
2023-05-01 20:37:13.807 INFO  c.k.k.c.keyword.builtin.CommentKeyword   - step5_backupLatest started
2023-05-01 20:37:14.001 ERROR c.k.katalon.core.main.TestCaseExecutor   - ❌ Test Cases/NISA/main_Chronos FAILED.
Reason:
com.kazurayam.inspectus.core.InspectusException: com.kazurayam.materialstore.core.MaterialstoreException: in /Users/kazuakiurayama/katalon-workspace/VisualInspectionOfExcelAndPDF/store, there is no dir of jobName=NISA_Chronos/jobTimestamp=_
	at com.kazurayam.inspectus.core.internal.AbstractService.step5_backupLatest(AbstractService.java:112)
	at com.kazurayam.inspectus.core.internal.AbstractService.postProcess(AbstractService.java:54)
	at com.kazurayam.inspectus.core.Inspectus.execute(Inspectus.java:13)
	at com.kazurayam.inspectus.core.Inspectus$execute.call(Unknown Source)
	at main_Chronos.run(main_Chronos:41)
	at com.kms.katalon.core.main.ScriptEngine.run(ScriptEngine.java:194)
	at com.kms.katalon.core.main.ScriptEngine.runScriptAsRawText(ScriptEngine.java:119)
	at com.kms.katalon.core.main.TestCaseExecutor.runScript(TestCaseExecutor.java:448)
	at com.kms.katalon.core.main.TestCaseExecutor.doExecute(TestCaseExecutor.java:439)
	at com.kms.katalon.core.main.TestCaseExecutor.processExecutionPhase(TestCaseExecutor.java:418)
	at com.kms.katalon.core.main.TestCaseExecutor.accessMainPhase(TestCaseExecutor.java:410)
	at com.kms.katalon.core.main.TestCaseExecutor.execute(TestCaseExecutor.java:285)
	at com.kms.katalon.core.main.TestCaseMain.runTestCase(TestCaseMain.java:144)
	at com.kms.katalon.core.main.TestCaseMain.runTestCase(TestCaseMain.java:135)
	at com.kms.katalon.core.main.TestCaseMain$runTestCase$0.call(Unknown Source)
	at TempTestCase1682940972842.run(TempTestCase1682940972842.groovy:25)
Caused by: com.kazurayam.materialstore.core.MaterialstoreException: in /Users/kazuakiurayama/katalon-workspace/VisualInspectionOfExcelAndPDF/store, there is no dir of jobName=NISA_Chronos/jobTimestamp=_
	at com.kazurayam.materialstore.base.manage.StoreExportImpl.exportJobTimestamps(StoreExportImpl.java:77)
	at com.kazurayam.materialstore.base.manage.StoreExportImpl.exportReports(StoreExportImpl.java:48)
	at com.kazurayam.inspectus.core.internal.AbstractService.step5_backupLatest(AbstractService.java:101)
	at com.kazurayam.inspectus.core.internal.AbstractService.postProcess(AbstractService.java:54)
	at com.kazurayam.inspectus.core.Inspectus.execute(Inspectus.java:13)
	at com.kazurayam.inspectus.core.Inspectus$execute.call(Unknown Source)
	at Script1682593110094.run(Script1682593110094.groovy:41)
	... 11 more

2023-05-01 20:37:14.011 ERROR c.k.katalon.core.main.TestCaseExecutor   - ❌ Test Cases/NISA/main_Chronos FAILED.
Reason:
com.kazurayam.inspectus.core.InspectusException: com.kazurayam.materialstore.core.MaterialstoreException: in /Users/kazuakiurayama/katalon-workspace/VisualInspectionOfExcelAndPDF/store, there is no dir of jobName=NISA_Chronos/jobTimestamp=_
	at com.kazurayam.inspectus.core.internal.AbstractService.step5_backupLatest(AbstractService.java:112)
	at com.kazurayam.inspectus.core.internal.AbstractService.postProcess(AbstractService.java:54)
	at com.kazurayam.inspectus.core.Inspectus.execute(Inspectus.java:13)
	at com.kazurayam.inspectus.core.Inspectus$execute.call(Unknown Source)
	at main_Chronos.run(main_Chronos:41)
	at com.kms.katalon.core.main.ScriptEngine.run(ScriptEngine.java:194)
	at com.kms.katalon.core.main.ScriptEngine.runScriptAsRawText(ScriptEngine.java:119)
	at com.kms.katalon.core.main.TestCaseExecutor.runScript(TestCaseExecutor.java:448)
	at com.kms.katalon.core.main.TestCaseExecutor.doExecute(TestCaseExecutor.java:439)
	at com.kms.katalon.core.main.TestCaseExecutor.processExecutionPhase(TestCaseExecutor.java:418)
	at com.kms.katalon.core.main.TestCaseExecutor.accessMainPhase(TestCaseExecutor.java:410)
	at com.kms.katalon.core.main.TestCaseExecutor.execute(TestCaseExecutor.java:285)
	at com.kms.katalon.core.main.TestCaseMain.runTestCase(TestCaseMain.java:144)
	at com.kms.katalon.core.main.TestCaseMain.runTestCase(TestCaseMain.java:135)
	at com.kms.katalon.core.main.TestCaseMain$runTestCase$0.call(Unknown Source)
	at TempTestCase1682940972842.run(TempTestCase1682940972842.groovy:25)
Caused by: com.kazurayam.materialstore.core.MaterialstoreException: in /Users/kazuakiurayama/katalon-workspace/VisualInspectionOfExcelAndPDF/store, there is no dir of jobName=NISA_Chronos/jobTimestamp=_
	at com.kazurayam.materialstore.base.manage.StoreExportImpl.exportJobTimestamps(StoreExportImpl.java:77)
	at com.kazurayam.materialstore.base.manage.StoreExportImpl.exportReports(StoreExportImpl.java:48)
	at com.kazurayam.inspectus.core.internal.AbstractService.step5_backupLatest(AbstractService.java:101)
	... 15 more

2023-05-01 20:37:14.015 INFO  c.k.katalon.core.main.TestCaseExecutor   - END Test Cases/NISA/main_Chronos

```

**Comment by kazurayam** [comment]
See the following screenshot.
<img width="1552" alt="スクリーンショット 2023-05-01 20 41 45" src="https://user-images.githubusercontent.com/4467417/235447094-deb33f8d-90fe-4b1d-b739-6af2c6e7a91c.png">

The `index` file contains 

```
"left":"_/0000000000000000000000000000000000000000"
```

This contains the JobTimestamp of "_".

This points a `Material.NULL_OBJECT` which is obviously not present on the file system.

**Comment by kazurayam** [comment]
JobTimestamp has the following code
:
```
    public static final String EPOCH_NAME = "_";

    public static final JobTimestamp NULL_OBJECT = new JobTimestamp(EPOCH_NAME);
```

**Comment by kazurayam** [comment]
In order to refrain from the exception raised at `StoreExportImple#exportJobTimestamps(StoreExportImpl.java:77)`, the `exportJobTimestamps()` method should explicitly exclude the JobTimestamp.NULL_OBJECT.

**Comment by kazurayam** [comment]
fixed by the materialstore-0.16.2

**Comment by kazurayam** [comment]
done at [v0.14.2-SNAPSHOT](https://github.com/kazurayam/VisualInspectionOfExcelAndPDF/releases/tag/0.14.2-SNAPSHOT)
